cordova-lib/src/cordova/restore-util.js

```diff
@@ -63,7 +64,27 @@ function installPlatformsFromConfigXML(platforms, opts) {
             // Combining arrays and checking duplicates.
             comboArray = pkgJsonPlatforms.slice();
         }
+
         engines = cfg.getEngines(projectHome);
+
+        if(fs.existsSync(pkgJsonPath)) {
+            pkgJson = require(pkgJsonPath);
+        // TODO: Eventually refactor out to pacakge manager module.
```

Comment:
Reminder to  ---> https://issues.apache.org/jira/browse/CB-12592. Want me to take it out?